packages/web-sdk/src/instrumentations/webVitalsWithAttribution/instrumentation.ts

```diff
@@ -0,0 +1,128 @@
+      this.addIfPresent(values, 'largest_shift_value', metric.attribution.largestShiftValue)
+      this.addIfPresent(values, 'largest_shift_time', metric.attribution.largestShiftTime)
+
+      const context = this.buildInitialContext(metric)
+      this.addIfPresent(context, 'load_state', metric.attribution.loadState)
+      this.addIfPresent(context, 'largest_shift_target', metric.attribution.largestShiftTarget)
+
+      this.pushMeasurement(values, context)
+    })
+  }
+
+  private measureFCP(): void {
+    onFCP((metric) => {
+      const values = this.buildInitialValues(metric)
+      this.addIfPresent(values, 'first_byte_to_fcp', metric.attribution.firstByteToFCP)
+      this.addIfPresent(values, 'time_to_first_byte', metric.attribution.timeToFirstByte)
+
+      const context = this.buildInitialContext(metric)
+      this.addIfPresent(context, 'load_state', metric.attribution.loadState)
+
+      this.pushMeasurement(values, context)
+    })
+  }
+
+  private measureFID(): void {
+    onFID((metric) => {
+      const values = this.buildInitialValues(metric)
+      this.addIfPresent(values, 'event_time', metric.attribution.eventTime)
+
+      const context = this.buildInitialContext(metric)
+      this.addIfPresent(context, 'event_target', metric.attribution.eventTarget)
+      this.addIfPresent(context, 'event_type', metric.attribution.eventType)
+      this.addIfPresent(context, 'load_state', metric.attribution.loadState)
+
+      this.pushMeasurement(values, context)
+    })
+  }
+
+  private measureINP(): void {
+    onINP((metric) => {
+      const values = this.buildInitialValues(metric)
+      this.addIfPresent(values, 'event_time', metric.attribution.eventTime)
+
+      const context = this.buildInitialContext(metric)
+      this.addIfPresent(context, 'event_target', metric.attribution.eventTarget)
+      this.addIfPresent(context, 'event_type', metric.attribution.eventType)
+      this.addIfPresent(context, 'load_state', metric.attribution.loadState)
+
+      this.pushMeasurement(values, context)
+    })
+  }
+
+  private measureLCP(): void {
+    onLCP((metric) => {
+      const values = this.buildInitialValues(metric)
+      this.addIfPresent(values, 'element_render_delay', metric.attribution.elementRenderDelay)
+      this.addIfPresent(values, 'resource_load_delay', metric.attribution.resourceLoadDelay)
+      this.addIfPresent(values, 'resource_load_time', metric.attribution.resourceLoadTime)
+      this.addIfPresent(values, 'time_to_first_byte', metric.attribution.timeToFirstByte)
+
+      const context = this.buildInitialContext(metric)
+      this.addIfPresent(context, 'element', metric.attribution.element)
+
+      this.pushMeasurement(values, context)
+    })
+  }
+
+  private measureTTFB(): void {
+    onTTFB((metric) => {
+      const values = this.buildInitialValues(metric)
+      this.addIfPresent(values, 'dns_time', metric.attribution.dnsTime)
```

Comment:
this difference is because of the version of web vitals that you [commented here](https://github.com/grafana/faro-web-sdk/pull/595#discussion_r1615928078). I will close the other threads on this subject to centralize the discussion